projects/vdk-core/src/vdk/internal/builtin_plugins/config/secrets_config.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2021-2024 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+
+
+class SecretsConfigPlugin:
+    @hookimpl(trylast=True)
+    def initialize_job(self, context: JobContext):
```

Comment:
This is a bit of a hack, not sure if it's a good idea. There was no way to mutate the configuration in an earlier hook, because it gets frozen once it gets built and there is no access to secrets, because the haven't been initialized yet. Also, importing JobContext in `vdk_config.py` causes a circular.